src/request.js

```diff
@@ -0,0 +1,237 @@
+let http = require('http');
+let https = require('https');
+let assert = require('assert');
+let urllib = require('url');
+let debug = require('debug')('requests');
+
+let versionString = '/unknownversion';
+
+// Figure out the version string that we'll include in the user agent.  If we
+// cannot figure out the version string, we'll (in the future) figure out what
+// git/hg revision we're on and use that.
+try {
```

Comment:
I was writing this in a different location at first.  I guess we don't really need it :/